lib/bump/dependency_file_updaters/ruby.rb

```diff
@@ -94,9 +95,18 @@ def build_updated_gemfile_lock
           end
         post_process_lockfile(lockfile_body)
       rescue SharedHelpers::ChildProcessFailed => error
-        raise unless error.error_class == "Bundler::VersionConflict"
+        handle_bundler_errors(error)
+      end
 
-        raise DependencyFileUpdaters::VersionConflict
+      def handle_bundler_errors(error)
+        case error.error_class
+        when "Bundler::VersionConflict"
+          raise DependencyFileUpdaters::VersionConflict
+        when "Bundler::Source::Git::GitCommandError"
+          raise Bump::GitDependencyNotFound
```

Comment:
Not really. We get one of these: https://github.com/bundler/bundler/blob/master/lib/bundler/source/git/git_proxy.rb#L26-L34, which unfortunately doesn't have details of the error raised (`error_msg` isn't what you'd expect it to be, and is always `nil`).

Could rename the error to `GitCommandError` and then leave the client to check existence? In future we could put the `command` onto `GitCommandError`, which would make it relatively straightforward to pull out the path.